File: 백준/Gold/11404. 플로이드/플로이드.go
package main

import (
	"bufio"
	"os"
	"strconv"
	"strings"
)

type BusLine struct {
	from int
	to   int
}

func main() {
	// Initialize

	reader := bufio.NewReader(os.Stdin)
	var line string

	line, _ = reader.ReadString('\n')
	n, _ := strconv.Atoi(strings.TrimSpace(line))

	line, _ = reader.ReadString('\n')
	m, _ := strconv.Atoi(strings.TrimSpace(line))

	busLines := make(map[BusLine]int, m)

	for i := 0; i < m; i++ {
		line, _ = reader.ReadString('\n')
		infoStrs := strings.Split(strings.TrimSpace(line), " ")
		from, _ := strconv.Atoi(infoStrs[0])
		to, _ := strconv.Atoi(infoStrs[1])
		cost, _ := strconv.Atoi(infoStrs[2])

		busLine := BusLine{from - 1, to - 1}

		if c, ok := busLines[busLine]; ok {
			if cost < c {
				busLines[busLine] = cost
			}
		} else {
			busLines[busLine] = cost
		}
	}

  // Floyd-Warshall
	dists := make([][]int, n)
	for i := 0; i < n; i++ {
		dists[i] = make([]int, n)
	}

  // connect direct
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			if j == i {
				continue
			}

			if c, ok := busLines[BusLine{i, j}]; 
			ok && (dists[i][j] == 0 || dists[i][j] > c) {
			  dists[i][j] = c
			}
		}
  }

  // connect indirect
	for i := 0; i < n; i++ {
    for h := 0; h < n; h++ {
      if h == i || dists[h][i] == 0 {
        continue
      }
      for j := 0; j < n; j++ {
        if j == h || j == i || dists[i][j] == 0 {
          continue
        }

        if dists[h][j] == 0 || dists[h][j] > dists[h][i] + dists[i][j] {
          dists[h][j] = dists[h][i] + dists[i][j]
        }
      }
    }
	}

	// Print output
  writer := bufio.NewWriter(os.Stdout)
  for _, distRow := range dists {
    for l, dist := range distRow {
      writer.WriteString(strconv.Itoa(dist))
      if l < len(distRow) - 1 {
        writer.WriteString(" ")
      } else {
        writer.WriteString("\n")
      }
    }
  }
  writer.Flush()
}

File: 백준/Gold/11404. 플로이드/플로이드.py
from typing import *
import sys
input = sys.stdin.readline
output = sys.stdout.write
flush = sys.stdout.flush
sys.setrecursionlimit(10**6)

n = int(input())
m = int(input())

# init graph
graph = [
    [0 if i == j else 100000*n for j in range(n+1)]
    for i in range(n+1)
]

for _ in range(m):
    f, t, c = map(int, input().split())
    graph[f][t] = min(c, graph[f][t])

# floyd warshall
for k in range(1, n+1):
    for a in range(1, n+1):
        for b in range(1, n+1):
            graph[a][b] = min(
                graph[a][b],
                graph[a][k] + graph[k][b]
            )

for i in range(1, n+1):
    for j in range(1, n+1):
        if graph[i][j] == 100000*n:
            graph[i][j] = 0

for i in range(1, n+1):
    for j in range(1, n+1):
        output(f"{graph[i][j]} ")
    output("\n")

flush()

